torch/_higher_order_ops/flex_attention.py

```diff
@@ -94,17 +91,23 @@ def _(
         return vals.new_empty(shape)
 
     @zeros_and_scatter.register_vmap  # type: ignore[misc]
-    def _(info, indims, shape, indices, val):  # type: ignore[no-untyped-def]
+    def _(info, indims, shape, indices, value):  # type: ignore[no-untyped-def]
         """The batching rule is special in that it returns a tensor that is not batched"""
-        if len(indices) > 1:
-            for i, idx in enumerate(indices):
-                # TODO: Don't use is_batchedtensor API, use in_dims instead.
-                if torch._C._functorch.is_batchedtensor(idx):
-                    indices[i] = idx.unsqueeze(-1)
+        indices_indims = indims[1]
+        expanded_indices = []
+        for idx, idx_indim in zip(indices, indices_indims):
+            # The index is not a being batched, we should unsqueeze and expand to val
+            if idx_indim is None:
+                expanded_indices.append(idx.unsqueeze(0).expand(value.shape))
```

Comment:
could simply `.append(idx.expand_as(value))` be used instead?

also maybe explicit `unsqueeze(0)` could be omitted? as broadcasting should already do it

Reply:
Good tip, thanks